extendr-api/src/iter.rs

```diff
@@ -89,14 +85,26 @@ impl Iterator for StrIter {
             let vector = self.vector.get();
             if i >= self.len {
                 None
-            } else if TYPEOF(vector) == SEXPTYPE::STRSXP {
-                Some(str_from_strsxp(vector, i as isize))
-            } else if TYPEOF(vector) == SEXPTYPE::INTSXP && TYPEOF(self.levels) == SEXPTYPE::STRSXP
-            {
-                let j = *(INTEGER(vector).add(i));
-                Some(str_from_strsxp(self.levels, j as isize - 1))
             } else if TYPEOF(vector) == SEXPTYPE::NILSXP {
                 Some(<&str>::na())
+            } else if TYPEOF(vector) == SEXPTYPE::STRSXP {
+                Some(str_from_strsxp(vector, i))
+            } else if TYPEOF(vector) == SEXPTYPE::CHARSXP {
+                if vector == R_NaString {
+                    Some(<&str>::na())
+                } else if vector == R_BlankString {
+                    Some("")
+                } else {
+                    Some(rstr::charsxp_to_str(vector))
+                }
```

Comment:
This fragment is duplicated

Reply:
You're right. I'll move things around and ping later.